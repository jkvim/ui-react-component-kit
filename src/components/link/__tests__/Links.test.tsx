import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Link } from '../';

describe('Link', () => {

  let component, callback;

  beforeEach(() => {
    callback = jest.fn();
    component = shallow(<Link onClick={callback} isExternal={true} url="http://somewhere" title="title" />);
  });

  test('snapshot', () => {
    expect(toJSON(component)).toMatchSnapshot();
  });

  test('callback', () => {
    expect(callback).not.toBeCalled();
    component.simulate('click');
    expect(callback).toBeCalled();
  });
});
