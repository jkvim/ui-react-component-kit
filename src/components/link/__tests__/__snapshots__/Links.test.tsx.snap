// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link snapshot 1`] = `
<styled.a
  href="http://somewhere"
  onClick={[MockFunction]}
  target="_blank"
  theme={
    Object {
      "color": Object {
        "default": "#2fa4e7",
        "disabled": "#acacac",
        "error": "#c71c22",
        "focus": "#54b4eb",
        "hover": "#178acc",
      },
      "font": Object {
        "size": "1.8rem",
      },
    }
  }
>
  title
  <styled.span>
    <styled.span>
      Link opens in new window
    </styled.span>
  </styled.span>
</styled.a>
`;
